server/api.js

```diff
@@ -4,15 +4,140 @@ import pool from "./db";
 
 const router = Router();
 
+// Get all the tasks
 router.get("/tasks", (req, res) => {
 	pool.query("SELECT * FROM tasks").then((result) => {
-		// console.log(result.rows);
 		return res.send(result.rows);
 	});
 });
-
 router.get("/", (_, res) => {
-	return res.json({ message: "Hello, from the api route!" });
+	res.json({ message: "Hello, from the api route!" });
 });
 
+router.post("/tasks", (req, res) => {
+	const { task_title, due_date, evidence, status_title, element_title } =
+		req.body;
+	let params = [];
+	if (!task_title) {
+		return res
+			.status(404)
+			.send({ success: false, message: "Task title cannot be empty" });
+	} else {
+		params.push(task_title);
+	}
+	if (due_date.length > 0) {
+		params.push(due_date);
+	} else {
+		params.push("");
+	}
+	if (evidence.length > 0) {
+		params.push(evidence);
+	} else {
+		params.push("");
+	}
+	if (status_title.length > 0) {
+		pool
+			.query("SELECT Status_id FROM status WHERE Status_title = $1", [
+				status_title,
+			])
```

Comment:
the status cannot be empty so it makes sense that we are not checking if the status_id returned is present, it will be present as its 1 of 4 options. But we must however account for when the server does not  send back anything with a catch block here too